pkgs/development/compilers/ghc/head.nix

```diff
@@ -12,7 +12,7 @@ let
 
   commonBuildInputs = [ ghc perl autoconf automake happy alex python3 ];
 
-  version = "8.1.20170106";
+  version = "2017-01-06";
```

Comment:
@rht, I don't understand what you're getting at. Could you please revert these changes? If you have a better suggestion to be used for GHC HEAD version numbers, then by all means please make a suggestion in the form of a new PR, but the version numbers we have now are unsuitable, IMHO.